sandbox/my_sod_shock_stube_solver/my_cese_solver.py

```diff
@@ -0,0 +1,186 @@
+        f32 = ga * w3 - a4 * w2**2      # gamma*u3/u1 - 1.5*(gamma-1.0)*(u2/u1)^2
+        f33 = ga * w2                   # gamma*u2/u1
+
+        # Eq.(4.17), (u_mt)nj = -(f_mx)nj = -(fm,k*u_kx)nj, (u_mt)nj -> qt, (u_kx)nj -> qx
+        mtx_qt[0][j] = -mtx_qx[1][j]
+        mtx_qt[1][j] = -(f21 * mtx_qx[0][j] + f22 * mtx_qx[1][j] + a1  * mtx_qx[2][j])
+        mtx_qt[2][j] = -(f31 * mtx_qx[0][j] + f32 * mtx_qx[1][j] + f33 * mtx_qx[2][j])
+
+        # Eq.(4.25), (u_m)nj -> q, (u_mt)nj -> qt, (u_kx)nj -> qx
+        # for s0: u_0x -> mtx_qx[0][j], f_0  -> mtx_q[1][j], f_0t -> f0,k*u_kt 
+        #     s1: u_1x -> mtx_qx[1][j], f_1  -> a1*mtx_q[2][j]+f21*mtx_q[0][j]+f22*mtx_q[1][j]
+        #                           f_1t -> f21*mtx_qx[0][j]+f22*mtx_qx[1][j]+a1*mtx_qx[2][j]
+        #     s2: u_2x -> mtx_qx[2][j], f_2  -> f31*mtx_q[0][j]+f32*mtx_q[1][j]+f33*mtx_q[2][j]
+        #                           f_2t -> f31*mtx_qx[0][j]+f32*mtx_qx[1][j]+f33*mtx_qx[2][j]
+        mtx_s[0][j] = qdx * mtx_qx[0][j] + dtx * (mtx_q[1][j] + qdt * mtx_qt[1][j])
+        
+        mtx_s[1][j] = qdx * mtx_qx[1][j] + dtx * (f21 * (mtx_q[0][j] + qdt * mtx_qt[0][j]) + \
+                                          f22 * (mtx_q[1][j] + qdt * mtx_qt[1][j]) + \
+                                          a1  * (mtx_q[2][j] + qdt * mtx_qt[2][j])) 
+        mtx_s[2][j] = qdx * mtx_qx[2][j] + dtx * (f31 * (mtx_q[0][j] + qdt * mtx_qt[0][j]) + \
+                                          f32 * (mtx_q[1][j] + qdt * mtx_qt[1][j]) + \
+                                          f33 * (mtx_q[2][j] + qdt * mtx_qt[2][j]))
+
+    # evaluate the status of gas after time stamp moves forward by 1 hdt
+    mm = m - 1
+    for j in xrange(0, mm): # j -> 1 hdx
+        for k in xrange(0, 3):
+            # Eq.(4.24), index of 'qn' will be the same for the following calculation
+            mtx_qn[k][j+1] = 0.5 * (mtx_q[k][j] + mtx_q[k][j+1] + mtx_s[k][j] - mtx_s[k][j+1])
+            # Eq.(4.27) and Eq.(4.36)  
+            vxl[k] = (mtx_qn[k][j+1] - mtx_q[k][j] - hdt * mtx_qt[k][j]) / hdx     # l -> -
+            vxr[k] = (mtx_q[k][j+1] + hdt * mtx_qt[k][j+1] - mtx_qn[k][j+1]) / hdx # r -> +
+            # Eq.(4.38) and Eq.(4.39)
+            mtx_qx[k][j+1] = (vxl[k] * (math.fabs(vxr[k]))**ia + vxr[k] * (math.fabs(vxl[k]))**ia)\
+                             / ((math.fabs(vxl[k]))**ia + (math.fabs(vxr[k]))**ia + 10**(-60))
+
+    for j in xrange(1, m):
+        for k in xrange(0, 3):
+            mtx_q[k][j] = mtx_qn[k][j]
+
+    if i % 2 != 0:
+        for j in xrange(1, m):
+            for k in xrange(0, 3):
+                mtx_q[k][j-1] = mtx_q[k][j]
+                mtx_qx[k][j-1] = mtx_qx[k][j]
+
+        for j in xrange(0, itp):
+            status_rho[j] = mtx_q[0][j]  # rho
+            status_vel[j] = mtx_q[1][j] / mtx_q[0][j]  # v
+            status_p[j] = a1 * (mtx_q[2][j] - 0.5 * mtx_q[0][j] * status_vel[j]**2) # p
+    
+
+        plt.subplot(311)
```

Comment:
This mixes solution calculation with visualization.  This style has drawbacks:

1. We can't disable visualization without changing the inner most loop.  

2. Because the plotting parameters are deep in the loop, configuring them will be difficult.

To improve, we should separate the calculation from other actions.  [A simple Lax-Wendroff solver](https://github.com/yungyuc/cythonup/blob/master/src/lax/lax/core.py) demonstrated how to do it.  More referential materials please find [here](https://github.com/yungyuc/cythonup/blob/master/module/interact_with_c/lax.ipynb).